### PR TITLE
Update and fix script

### PR DESCRIPTION
**Pull request by kyessenov**
Previous version had incorrect reference to locally built protoc-gen-validate.
Use the system version in Dockerfile and set the version correctly to last working version of protoc-gen-validate.

/cc @yangminzhu

### CODE REVIEW

**Comment by kyessenov** [comment]
Fixes #193

**Comment by kyessenov** [comment]
@rshriram can you bless so we can move data-plane-api forward in istio?

File: Dockerfile.ci [hunk 1]
```diff
@@ -9,7 +9,7 @@ RUN go mod download
 
 # Fetch protoc modules
 RUN go get github.com/gogo/protobuf/protoc-gen-gogofast@v1.2.1
-RUN go get github.com/envoyproxy/protoc-gen-validate@v0.0.14
+RUN go get github.com/envoyproxy/protoc-gen-validate@d6164de4910977d3c3c8dbd9299b5064ea9e7c2b
```
Comment (yangminzhu) [inline Dockerfile.ci]: Is it possible to use the latest sha? It seems I'm having some issues with this sha after updating the `data-plane-api` to latest:
```bash
$ docker run -v $(pwd):/go-control-plane gcr.io/istio-testing/go-control-plane-ci:07-08-2019 make generate
--> vendoring protobufs
go: finding github.com/envoyproxy/data-plane-api v0.0.0-20190627184114-829b48c237a2
go: downloading github.com/envoyproxy/data-plane-api v0.0.0-20190627184114-829b48c237a2
go: extracting github.com/envoyproxy/data-plane-api v0.0.0-20190627184114-829b48c237a2
--> generating pb.go files
/bin/bash build/generate_protos.sh
Generating protos /go-control-plane/build/../data-plane-api/envoy/type ...
/go-control-plane/build/../vendor/github.com/envoyproxy/protoc-gen-validate: warning: directory does not exist.
/go-control-plane/build/../vendor/github.com/gogo/googleapis: warning: directory does not exist.
/go-control-plane/build/../vendor/istio.io/gogo-genproto/prometheus: warning: directory does not exist.
/go-control-plane/build/../vendor/istio.io/gogo-genproto: warning: directory does not exist.
validate/validate.proto: File not found.
envoy/type/http_status.proto: Import "validate/validate.proto" was not found or had errors.
Makefile:65: recipe for target 'generate' failed
make: *** [generate] Error 1
```

Comment (kyessenov) [reply Dockerfile.ci]: Yes, I can't use the latest sha or even 0.1.0. There seems to be a bug in the upstream. Probably worth fixing it, but we don't need to block the update by that.

Comment (yangminzhu) [reply Dockerfile.ci]: Also I cannot build the Docker image with any of the sha, it seems it's not cached by gocenter.io:
```bash
$ docker build -f Dockerfile.ci -t my-control-plane-api:v1 .
Sending build context to Docker daemon  75.43MB
Step 1/12 : FROM golang:1.12.5-stretch
 ---> 1ef078f0da9e
Step 2/12 : WORKDIR /go-control-plane
 ---> Using cache
 ---> ba45bea77c37
Step 3/12 : ENV GOPROXY=https://gocenter.io
 ---> Using cache
 ---> ea389517aa9a
Step 4/12 : COPY go.mod ./
 ---> Using cache
 ---> c20d13cfa883
Step 5/12 : RUN go mod download
 ---> Using cache
 ---> 92f2e17cd37b
Step 6/12 : RUN go get github.com/gogo/protobuf/protoc-gen-gogofast@v1.2.1
 ---> Using cache
 ---> d20ea4af8e70
Step 7/12 : RUN go get github.com/envoyproxy/protoc-gen-validate@d6164de4910977d3c3c8dbd9299b5064ea9e7c2b
 ---> Running in 02e56305677a
go: finding github.com/envoyproxy/protoc-gen-validate d6164de4910977d3c3c8dbd9299b5064ea9e7c2b
go: finding github.com/envoyproxy d6164de4910977d3c3c8dbd9299b5064ea9e7c2b
go: finding github.com d6164de4910977d3c3c8dbd9299b5064ea9e7c2b
go get github.com/envoyproxy/protoc-gen-validate@d6164de4910977d3c3c8dbd9299b5064ea9e7c2b: unexpected status (https://gocenter.io/github.com/envoyproxy/protoc-gen-validate/@v/d6164de4910977d3c3c8dbd9299b5064ea9e7c2b.info): 404 Not Found
The command '/bin/sh -c go get github.com/envoyproxy/protoc-gen-validate@d6164de4910977d3c3c8dbd9299b5064ea9e7c2b' returned a non-zero code: 1
```

Comment (kyessenov) [reply Dockerfile.ci]: That's weird. I just built this locally.

Comment (yangminzhu) [reply Dockerfile.ci]: Beyond unable to build the Docker image, I'm still seeing the error `validate/validate.proto: File not found.` with image `07-08-2019`

Comment (kyessenov) [reply Dockerfile.ci]: You should be able to run all the commands in dockerfile on your machine and get the same results. Have you tried that? Make sure your permissions are not accidentally set to root due to docker run.

Comment (kyessenov) [reply Dockerfile.ci]: Updated data-plane-api to latest @yangminzhu

**Pull request closed without merging**